circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/util/NameServicePathResolver.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hotels.bdp.circustrain.avro.util;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DFSConfigKeys;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+public class NameServicePathResolver {
+  private static final Logger LOG = LoggerFactory.getLogger(NameServicePathResolver.class);
+
+  private final Configuration configuration;
+
+  public NameServicePathResolver(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  public Path resolve(String url) {
+    String nameService = configuration.get(DFSConfigKeys.DFS_NAMESERVICES);
```

Comment:
Is this something we need to document? i.e. how you set this up and when it's used?

Reply:
I'm not sure, Hadoop docs cover it quite well, and the users that reported the issue were already aware of the configuration

Reply:
OK, so is this something they've set already in one of the Hadoop config files that is on the classpath? i.e. it's not something they set in Circus Train config anywhere?

Reply:
No only `dfs.nameservices`

Reply:
I think this is the minimum config required for copy to succeed

`source-catalog:
...
configuration-properties:
dfs.nameservices: foo-ha
dfs.client.failover.proxy.provider.foo-ha: org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider
dfs.ha.namenodes.foo-ha: nn1,nn2
dfs.namenode.rpc-address.foo-ha.nn1: nn1host:8020
dfs.namenode.rpc-address.foo-ha.nn2: nn2host:8020`

but circus-train-avro specifically only needs `dfs.nameservices`